### PR TITLE
DOCSUP-5909: Documented array aggregation functions

### PR DESCRIPTION
**Pull request by gyuton**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category :
- Documentation for #18032

### CODE REVIEW

File: docs/ru/sql-reference/functions/array-functions.md [hunk 1]
```diff
@@ -1135,11 +1135,225 @@ SELECT
...
+
+Функция `arrayMin` является [функцией высшего порядка](../../sql-reference/functions/index.md#higher-order-functions) — в качестве первого аргумента ей можно передать лямбда-функцию.
+
+**Синтаксис**
+
+```sql
+arrayMin([func,] arr)
+```
+
+**Parameters**
```
Comment (olgarev) [inline docs/ru/sql-reference/functions/array-functions.md:1152]:
```suggestion
**Параметры**
```

**Comment by olgarev** [comment]
@Avogar Hi, will you review this, pls?

File: docs/en/sql-reference/functions/array-functions.md [hunk 2]
```diff
@@ -1288,73 +1288,226 @@ Returns the index of the first element in the `arr1` array for which `func` retu
...
+**Parameters**
+
+-   `func` — Function. [Expression](../../sql-reference/data-types/special-data-types/expression.md).
+-   `arr` — Array. [Array](../../sql-reference/data-types/array.md).
+
+**Returned value**
+
+-   The minimum of function values (or the array minimum). 
+
+Type: matches the array elements type.
```
Comment (Avogar) [inline docs/en/sql-reference/functions/array-functions.md:1314]: That's not correct, if `func` is specified, `arrayMin` return value type matches `func` return value type.
Example:
```
SELECT arrayMin(x -> toFloat32(x), [1,2,3])
```
The result value will have type Float32.

File: docs/en/sql-reference/functions/array-functions.md [hunk 3]
```diff
@@ -1288,73 +1288,226 @@ Returns the index of the first element in the `arr1` array for which `func` retu
...
+**Parameters**
+
+-   `func` — Function. [Expression](../../sql-reference/data-types/special-data-types/expression.md).
+-   `arr` — Array. [Array](../../sql-reference/data-types/array.md).   
+
+**Returned value**
+
+-   The sum of the function values (or the array sum).
+
+Type: for decimal numbers in source array — [Decimal128](../../sql-reference/data-types/decimal.md), for floating point numbers — [Float64](../../sql-reference/data-types/float.md), for numeric unsigned — [UInt64](../../sql-reference/data-types/int-uint.md), and for numeric signed — [Int64](../../sql-reference/data-types/int-uint.md).
```
Comment (Avogar) [inline docs/en/sql-reference/functions/array-functions.md:1424]: If `func` is specified, this rule will be applied to `func` return value type, not to array elements type.

**Review by Avogar — APPROVED** [review]

**Pull request merged**